### PR TITLE
Add nccl support to MASGD.

### PR DESCRIPTION
**Pull request by nluehr**

### CODE REVIEW

File: Source/SGDLib/MASGD.h [hunk 1]
```diff
@@ -388,18 +393,29 @@ namespace Microsoft { namespace MSR { namespace CNTK {
...
+                    m_pMPI->AllReduce(px.get(), nx);
+                    commTimer.Stop();
+                    secondsOnCommunication += (float)commTimer.ElapsedSeconds();
+                    // 2.1.5. set value
+                    pNode->Value().SetValue(mat.GetNumRows(), mat.GetNumCols(), mat.GetDeviceId(), px.get());
+                    // 2.1.6. clean up
+                    //delete[]px;
+                }
+                else
+	        {
```
Comment (junjieqian) [inline Source/SGDLib/MASGD.h]: Replace Tab with space.

File: Source/SGDLib/MASGD.h [hunk 2]
```diff
@@ -388,18 +393,29 @@ namespace Microsoft { namespace MSR { namespace CNTK {
...
+                    commTimer.Restart();
+                    m_nccl.AllReduce(mat);
+                    commTimer.Stop();
+                    secondsOnCommunication += (float)commTimer.ElapsedSeconds();
+                }
```
Comment (junjieqian) [inline Source/SGDLib/MASGD.h]: Do we need m_nccl.Sync() here to sync all ops?

Comment (nluehr) [inline Source/SGDLib/MASGD.h]: Yes. Thanks.

File: Source/SGDLib/MASGD.h (hunk 1 above)
Comment (nluehr) [inline Source/SGDLib/MASGD.h]: Done.

**Comment by junjieqian** [comment]
I made some minor changes to catch up with master. It is in master now, https://github.com/Microsoft/CNTK/commit/c74d15c7c03cc64b1828cd9bb076f2bfde2956a8.
Thank you for your contribution!

**Pull request closed without merging**